crates/common/chain/lean/src/lean_chain.rs

```diff
@@ -207,33 +212,33 @@ impl LeanChain {
                 lean_block_provider.get(self.safe_target)?.ok_or_else(|| {
                     anyhow!("Block not found for safe target hash: {}", self.safe_target)
                 })?;
-            if target_block.message.slot > safe_target_block.message.slot {
+            if target_block.message.block.slot > safe_target_block.message.block.slot {
```

Comment:
```suggestion
            if target_block.slot > safe_target_block.slot {
```

Suggestion for better readability. All variables end with `_block` are actually typed `SignedBlockWithAttestation`, so I think we can access `.message.block` at assigning it. For example, `safe_target_block` can be assigned like:

```rust
let safe_target_block = lean_block_provider
    .get(self.safe_target)?
    .ok_or_else(|| {
        anyhow!("Block not found for safe target hash: {}", self.safe_target)
    })?
    .message
    .block;
```

What do you think? If this seems good, it would be best if we can apply this kind of changes all over this PR as this PR already modifies those lines.